commit 638d39f1d810c8f5e85d91f34dfa577308567bad
Author: Jens Lody <jens@codeblocks.org>
Date:   Wed Jul 25 22:49:34 2012 +0200

    wxSTC: revert workaround related to #9057, because it's not needed any longer

Index: include/wx/stc/stc.h
===================================================================
--- include/wx/stc/stc.h
+++ include/wx/stc/stc.h
@@ -4291,9 +4291,6 @@ protected:
 
     bool                m_lastKeyDownConsumed;
 
-    // Time until when we should ignore any new mouse wheel events.
-    wxLongLong m_timeToBlockWheelEventsUntil;
-
     friend class ScintillaWX;
     friend class Platform;
 #endif // !SWIG
Index: src/stc/stc.cpp
===================================================================
--- src/stc/stc.cpp
+++ src/stc/stc.cpp
@@ -199,7 +199,6 @@ bool wxStyledTextCtrl::Create(wxWindow *parent,
     m_swx = new ScintillaWX(this);
     m_stopWatch.Start();
     m_lastKeyDownConsumed = false;
-    m_timeToBlockWheelEventsUntil = 0;
     m_vScrollBar = NULL;
     m_hScrollBar = NULL;
 #if wxUSE_UNICODE
@@ -4258,25 +4257,11 @@ void wxStyledTextCtrl::OnContextMenu(wxContextMenuEvent& evt) {
 
 void wxStyledTextCtrl::OnMouseWheel(wxMouseEvent& evt)
 {
-    // Prevent having an event queue with wheel events that cannot be processed
-    // reasonably fast (see ticket #9057) by ignoring all of them that happen
-    // during the time interval corresponding to the time it took us to handle
-    // the last one.
-    //
-    // Notice the use of TimeInMicro() instead of Time() to avoid overflow in
-    // long running programs.
-    if ( m_timeToBlockWheelEventsUntil <= m_stopWatch.TimeInMicro() )
-    {
-        const wxLongLong beforeMouseWheel = m_stopWatch.TimeInMicro();
-        m_swx->DoMouseWheel(evt.GetWheelRotation(),
-                            evt.GetWheelDelta(),
-                            evt.GetLinesPerAction(),
-                            evt.ControlDown(),
-                            evt.IsPageScroll());
-        const wxLongLong afterMouseWheel = m_stopWatch.TimeInMicro();
-        m_timeToBlockWheelEventsUntil = afterMouseWheel;
-        m_timeToBlockWheelEventsUntil += afterMouseWheel - beforeMouseWheel;
-    }
+	m_swx->DoMouseWheel(evt.GetWheelRotation(),
+						evt.GetWheelDelta(),
+						evt.GetLinesPerAction(),
+						evt.ControlDown(),
+						evt.IsPageScroll());
 }
 
 
